src/client/async_std.rs

```diff
@@ -67,6 +67,15 @@ impl Client {
         client
     }
 
+    /// Clones a new client with idempotency key headers.
+    ///
+    /// For short living idempotency keys, they only matter on push requests
+    pub fn with_idempotency_key(&self, idempotency_key: String) -> Client {
+        let mut client = self.clone();
```

Comment:
I would suggest allowing the user to make the decision to clone, and instead move `self`. That way the user can re-use the same object. Something along the lines of this (note we take self, not a borrow):

```rust
pub fn with_idempotency_key(self, ) -> Self {
  Self {
    // update params with key
    ..self
  }
}

Reply:
Then in the user facing api it looks like this:

```rust
let client = ...
let idempotent_client = client.clone().with_idempotency_key("test".to_string());
// here both clients are valid
// alternatively
let idempotent_client = client.with_idempotency_key("test".to_string());
// now we 'move' the original client